src/dependencies.rs

```diff
@@ -80,20 +80,24 @@ pub fn build_dependencies(config: &mut Config) -> Result<Dependencies> {
     let artifact_output = output.stdout;
     let artifact_output = String::from_utf8(artifact_output)?;
     let mut import_paths: HashSet<PathBuf> = HashSet::new();
-    let mut artifacts: HashMap<_, _> = artifact_output
-        .lines()
-        .filter_map(|line| {
-            let message = serde_json::from_str::<cargo_metadata::Message>(line).ok()?;
-            if let cargo_metadata::Message::CompilerArtifact(artifact) = message {
-                for filename in &artifact.filenames {
-                    import_paths.insert(filename.parent().unwrap().into());
-                }
-                Some((artifact.package_id, artifact.filenames))
-            } else {
-                None
+    let mut artifacts = HashMap::new();
+    for line in artifact_output.lines() {
+        let Ok(message) = serde_json::from_str::<cargo_metadata::Message>(line) else {
+            continue
+        };
+        if let cargo_metadata::Message::CompilerArtifact(artifact) = message {
+            for filename in &artifact.filenames {
+                import_paths.insert(filename.parent().unwrap().into());
```

Comment:
Should this assert that no existing entry is overwritten?

Reply:
No, this doesn't matter, it is just about import paths, and these are the same for most dependencies.